lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt

```diff
@@ -250,10 +252,36 @@ class DefaultThumbnailsEngine(
             positions.mapNotNull { (positionUs, request) ->
                 val localizedUs = timer.localize(TrackType.VIDEO, index, positionUs)
                 localizedUs?.let { Stub(request, positionUs, localizedUs) }
-            }.toMutableList().sortedBy { it.positionUs }
+//            }.toMutableList().sortedBy { it.positionUs }
+            }.toMutableList().reorder(dataSources[TrackType.VIDEO][0].keyFrameTimestampsUs)
         )
     }
 
+    private fun  List<Stub>.reorder(keyFrameTimestampsUs: ArrayList<Long>): Collection<Stub> {
+        if(keyFrameTimestampsUs.isEmpty())
+            return this
+        bucketListMap.clear()
+        finalList.clear()
+
+        forEach {
+            val keyFrame = findPrevKeyFrame(it.positionUs, keyFrameTimestampsUs)
```

Comment:
findPrevKeyFrame can return -1.
Either we stop returning -1 from there..  or Handle -1 here, whichever is better